torch/distributions/transforms.py

```diff
@@ -412,6 +412,29 @@ def _inverse(self, y):
         return probs.log()
 
 
+class InvertableBoltzmannTransform(Transform):
+    """
+    Transform from unconstrained space to the simplex via `y = [exp(x), 1]` and
+    then normalizing. Note that dimensionality of `y.dim() == x.dim() + 1`.
+    """
+    domain = constraints.real
+    codomain = constraints.simplex
+    event_dim = 1
+
+    def __eq__(self, other):
+        return isinstance(other, InvertableBoltzmannTransform)
+
+    def _call(self, x):
+        ones = torch.ones(*(x.size()[:-1] + (1, )))
+        probs = torch.cat([x.exp(), ones], dim=-1)
+        probs /= probs.sum(-1, True)
+        return probs
+
+    def _inverse(self, y):
+        logprobs = y[:, :-1].log() - y[:, -1:].log()
+        return logprobs
```

Comment:
can you compress this to a single `return`? the assignment is unnecessary

Reply:
Done.